packages/forms-web-app/__tests__/unit/controllers/register/start.test.js

```diff
@@ -1,44 +1,103 @@
 const startController = require('../../../../src/controllers/register/start');
-const { VIEW } = require('../../../../src/lib/views');
+const { getAppData } = require('../../../../src/services/application.service');
 const { mockReq, mockRes } = require('../../mocks');
 
 jest.mock('../../../../src/lib/logger');
 
+jest.mock('../../../../src/services/application.service');
+
 describe('controllers/register/start', () => {
-  let req;
   let res;
+  let responseWithStatus;
 
   beforeEach(() => {
-    req = mockReq();
-    res = mockRes();
     jest.resetAllMocks();
+    res = mockRes();
+    responseWithStatus = mockRes();
+    res.status.mockImplementation(() => responseWithStatus);
   });
 
   describe('getStart', () => {
-    it('should call the correct template if caseRef is provided', () => {
-      req = {
+    it('should load project data and return register start view', async () => {
+      getAppData.mockImplementation(() =>
+        Promise.resolve({
+          resp_code: 200,
+          data: {
+            ProjectName: 'St James Barton Giant Wind Turbine',
+            DateOfRelevantRepresentationClose: '2024-09-01',
+          },
+        })
+      );
+      const req = {
         ...mockReq(),
+        params: {
+          case_ref: 'ABC123',
+        },
+      };
+      await startController.getStart(req, res);
+      expect(res.render).toHaveBeenCalledWith('register/start', {
+        projectName: 'St James Barton Giant Wind Turbine',
+        closeDate: 'Sunday 01 September 2024',
+      });
+    });
+    it('should load project data and return register start view even when existing project in session  ', async () => {
+      getAppData.mockImplementation(() =>
+        Promise.resolve({
+          resp_code: 200,
+          data: {
+            ProjectName: 'St James Barton Giant Wind Turbine',
+            DateOfRelevantRepresentationClose: '2024-09-01',
+          },
+        })
+      );
+      const req = {
+        ...mockReq(),
+        params: {
+          case_ref: 'ABC123',
+        },
         session: {
```

Comment:
I think we don't need session any more and can be removed from here

Reply:
Hey @pk2453 - in this scenario I kept the session to prove that any existing session value will not be used but rather the project details will be loaded based on the incoming case_ref in the URL path.  The scenario title is:

should load project data and return register start view even when existing project in session